internal/v4/api.go

```diff
@@ -222,6 +223,37 @@ func (h *Handler) updateEntity(id *charm.Reference, fields map[string]interface{
 	return nil
 }
 
+func (h *Handler) updateSearch(id *charm.Reference, fields map[string]interface{}) error {
+	return h.store.UpdateSearch(id)
+}
+
+func (h *Handler) updateSearchBase(id *charm.Reference, fields map[string]interface{}) error {
+	it := h.store.DB.Entities().Find(bson.M{
+		"name": id.Name,
+		"user": id.User,
+	}).Iter()
+	defer it.Close()
+	updated := make(map[string]bool)
+	var entity mongodoc.Entity
+	for it.Next(&entity) {
+		url := &charm.Reference{
```

Comment:
Just set the revision to -1 in a copy of entity.URL?
Also, could you please add a comment explaining why we are doing that?
I suppose that's because we only update the last revision, but a comment would help.